### PR TITLE
boards: add support for integer board revisions

### PR DESCRIPTION
**Pull request by henrikbrixandersen**
Add support for simple integer board revisions (1, 2, 3, ...).

Signed-off-by: Henrik Brix Andersen <hebad@vestas.com>

### CODE REVIEW

File: doc/guides/porting/board_porting.rst [hunk 1]
```diff
@@ -538,9 +538,10 @@ When the user builds for board ``plank@<revision>``:
   application for the board.
 
 Currently, ``<revision>`` can be either a numeric ``MAJOR.MINOR.PATCH`` style
-revision like ``1.5.0``, or single letter like ``A``, ``B``, etc. Zephyr
-provides a CMake board extension function, ``board_check_revision()``, to make
-it easy to match either style from :file:`revision.cmake`.
+revision like ``1.5.0``, an integer number, or single letter like ``A``, ``B``,
```
Comment (gmarull) [inline doc/guides/porting/board_porting.rst:541]:
```suggestion
revision like ``1.5.0``, an integer number like ``1``, or single letter like ``A``, ``B``,
```

**Review by gmarull — APPROVED** [review]

File: doc/guides/porting/board_porting.rst (hunk 1 above)
Comment (henrikbrixandersen) [reply doc/guides/porting/board_porting.rst]: Thanks, added.

**Review by SebastianBoe — APPROVED** [review]

**Review by tejlmand — CHANGES_REQUESTED** [review]
This is actually already possible as trailing zeroes can be omitted.

See here for show case: 
https://github.com/tejlmand/zephyr/commit/793024ab219ddf80a047652a91c65739ec21463a

That allows to build for single digit like:
```
$ cmake -GNinja -DBOARD=qemu_cortex_m3 -Bbuild samples/hello_world/
...
-- Board: qemu_cortex_m3, Revision: 1.0.0 (Active: 1)
```
or revision `2`:
```
$ cmake -GNinja -DBOARD=qemu_cortex_m3@2 -Bbuild samples/hello_world/
...
-- Board: qemu_cortex_m3, Revision: 2.0.0 (Active: 2)
```

**Comment by tejlmand** [comment]
> I have seen that feature, but it does not produce the same result. @2 is just a shorthand for @2.0.0 which will be what BOARD_REVISION is set to. With the NUMBER format in this PR, BOARD_REVISION will be set to 2.

Good point.

**Review by tejlmand — APPROVED** [review]

**Pull request merged**